src/main/java/ch/bfh/bti7081/s2018/green/views/MedicationView.java

```diff
@@ -1,71 +1,90 @@
 package ch.bfh.bti7081.s2018.green.views;
 
-import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Locale;
-
 import com.vaadin.navigator.View;
+import com.vaadin.ui.Button;
 import com.vaadin.ui.CustomLayout;
 import com.vaadin.ui.Grid;
-
 import ch.bfh.bti7081.s2018.green.models.entities.Medication;
 import ch.bfh.bti7081.s2018.green.presenters.MedicationPresenter;
+import com.vaadin.ui.renderers.LocalDateTimeRenderer;
+import com.vaadin.ui.renderers.Renderer;
+import org.vaadin.teemu.switchui.Switch;
 
 
 public class MedicationView extends CustomLayout implements View {
 
     public static final String NAME = "medication";
-    
-    private NavigationButton btnAddMedication;
+
+    private Button btnAddMedication;
     private Grid<Medication> grdMedicamentGridView = new Grid<>();
-  
-    public MedicationView() {
-        this.setTemplateName("medicationApp");
-        buildView();
-        new MedicationPresenter(this);
+
+    private Switch showExpired = new Switch("",false);
+    public Switch getShowExpired() {
+        return showExpired;
     }
 
-    public MedicationView(List<Medication> medDataList) {
-        this.setTemplateName("medicationApp");
+    public MedicationView() {
+        this.setTemplateName("medication");
+        this.setId(NAME);
         buildView();
+        if (this.showExpired.getValue())
+        {
+            new MedicationPresenter(this, true);
```

Comment:
Please add a getter in the view (that returns the value of the filter) and get it from the presenter